pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/readerwriter/ValueReaderComparisonTest.java

```diff
@@ -145,8 +156,12 @@ void testSuperiorPrefixes(ValueReader reader, int numBytesPerValue, String... st
           if (!Character.isLowSurrogate(chars[j])) {
             chars[j]++;
             String string = new String(chars);
-            int signum = strings[i].compareTo(string);
-            assertUtf8Comparison(reader, i, numBytesPerValue, string, Integer.compare(signum, 0));
+            // we can only compare if the string is the same length as the stored value.
+            // String constructor may compress the char array.
+            if (string.getBytes(StandardCharsets.UTF_8).length == numBytesPerValue) {
```

Comment:
This is not the correct check. The input value size won't match `numBytesPerValue`